### PR TITLE
Improve error message for missing operators

### PR DESCRIPTION
**Pull request by electrum**

### CODE REVIEW

File: presto-main/src/test/java/io/prestosql/type/TestRowOperators.java [hunk 1]
```diff
@@ -481,7 +481,7 @@ public void testRowComparison()
         assertFunction("row(2, CAST(NULL AS INTEGER)) = row(1, 2)", BOOLEAN, false);
         assertFunction("row(2, CAST(NULL AS INTEGER)) != row(1, 2)", BOOLEAN, true);
         assertInvalidFunction("row(TRUE, ARRAY [1, 2], MAP(ARRAY[1, 3], ARRAY[2.0E0, 4.0E0])) > row(TRUE, ARRAY [1, 2], MAP(ARRAY[1, 3], ARRAY[2.0E0, 4.0E0]))",
-                TYPE_MISMATCH, "line 1:64: '>' cannot be applied to row(boolean, array(integer), map(integer, double)), row(boolean, array(integer), map(integer, double))");
+                TYPE_MISMATCH, "line 1:64: Operator does not exist: row(boolean, array(integer), map(integer, double)) > row(boolean, array(integer), map(integer, double))");
```
Comment (findepi) [inline presto-main/src/test/java/io/prestosql/type/TestRowOperators.java:484]: Wasn't the original wording more readable?

Comment (findepi) [reply presto-main/src/test/java/io/prestosql/type/TestRowOperators.java]: `integer = text` looks nice. For complex types (like the one here), it's hard to fish out the actual operator... maybe it's hard no matter what we do though.

File: presto-main/src/main/java/io/prestosql/metadata/OperatorNotFoundException.java [hunk 2]
```diff
@@ -52,17 +52,34 @@ public OperatorNotFoundException(OperatorType operatorType, List<? extends Type>
...
-                operatorString = format("%s%s", operatorType.getOperator(), returnType.map(value -> " to " + value).orElse(""));
-                break;
-            default:
-                operatorString = format("'%s'%s", operatorType.getOperator(), returnType.map(value -> ":" + value).orElse(""));
+                return format("Cannot cast %s to %s", argumentTypes.get(0), returnType.orElseThrow());
+            case SUBSCRIPT:
+                return format("Cannot use %s for subscript of %s", argumentTypes.get(1), argumentTypes.get(0));
         }
-        return format("%s cannot be applied to %s", operatorString, Joiner.on(", ").join(argumentTypes));
+        return format("Operator '%s'%s cannot be applied to %s",
```
Comment (findepi) [inline presto-main/src/main/java/io/prestosql/metadata/OperatorNotFoundException.java:79]:
```suggestion
        return format(
            "Operator '%s'%s cannot be applied to %s",
```

File: presto-main/src/main/java/io/prestosql/metadata/OperatorNotFoundException.java [hunk 3]
```diff
@@ -52,17 +52,34 @@ public OperatorNotFoundException(OperatorType operatorType, List<? extends Type>
...
+                operatorType.getOperator(),
+                returnType.map(value -> ":" + value).orElse(""),
```
Comment (findepi) [inline presto-main/src/main/java/io/prestosql/metadata/OperatorNotFoundException.java:81]: Do we have a test case which would hit the `returnType.isPresent()` case?

File: presto-main/src/main/java/io/prestosql/metadata/OperatorNotFoundException.java [hunk 4]
```diff
@@ -52,17 +52,34 @@ public OperatorNotFoundException(OperatorType operatorType, List<? extends Type>
...
+            case LESS_THAN_OR_EQUAL:
+            case GREATER_THAN:
+            case GREATER_THAN_OR_EQUAL:
+                return format("Cannot apply operator: %s %s %s", argumentTypes.get(0), operatorType.getOperator(), argumentTypes.get(1));
+            case NEGATION:
+                return format("Cannot negate %s", argumentTypes.get(0));
             case BETWEEN:
                 return format("Cannot check if %s is BETWEEN %s and %s", argumentTypes.get(0), argumentTypes.get(1), argumentTypes.get(2));
+            case IS_DISTINCT_FROM:
+                return format("Cannot check if %s IS DISTINCT FROM %s", argumentTypes.get(0), argumentTypes.get(1));
```
Comment (findepi) [inline presto-main/src/main/java/io/prestosql/metadata/OperatorNotFoundException.java:73]: lowercase IS DISTINCT FROM ?
(you didn't upper cast in the CAST case; and the sentence is still a sentence when you treat "is distinct from" as just words)

Comment (electrum) [reply presto-main/src/main/java/io/prestosql/metadata/OperatorNotFoundException.java:73]: I was following BETWEEN, but I'll change that as well.

File: presto-main/src/main/java/io/prestosql/metadata/OperatorNotFoundException.java (hunk 3 above)
Comment (electrum) [reply presto-main/src/main/java/io/prestosql/metadata/OperatorNotFoundException.java:81]: I'm not aware of a way for a query to hit any of the other operators here.

File: presto-main/src/main/java/io/prestosql/metadata/OperatorNotFoundException.java (hunk 4 above)
Comment (electrum) [reply presto-main/src/main/java/io/prestosql/metadata/OperatorNotFoundException.java:73]: Actually, this matches what is used in `ExpressionAnalyzer`, so I'll make it `DISTINCT FROM`.

**Pull request merged**